qa/restart-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/SemanticSearch.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.bwc;
+
+import com.carrotsearch.randomizedtesting.RandomizedTest;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.ArrayList;
+
+
+import static org.opensearch.neuralsearch.TestUtils.NODES_BWC_CLUSTER;
+
+import static org.opensearch.neuralsearch.TestUtils.getModelId;
+import static org.opensearch.neuralsearch.TestUtils.TEXT_EMBEDDING_PROCESSOR;
+import org.opensearch.neuralsearch.query.NeuralQueryBuilder;
+
+
+public class SemanticSearch extends AbstractRestartUpgradeRestTestCase{
+
+    private static final String PIPELINE_NAME = "nlp-pipeline";
+    private static String DOC_ID = "0";
+    private static final String TEST_FIELD = "passage_text";
+    private static final String TEXT= "Hello world";
+
+    //Test restart-upgrade Semantic Search
+    //Create Text Embedding Processor, Ingestion Pipeline and add document
+    //Validate process , pipeline and document count in restart-upgrade scenario
+    public void testSemanticSearch() throws Exception{
+        waitForClusterHealthGreen(NODES_BWC_CLUSTER);
+
+        if (isRunningAgainstOldCluster()){
+            String modelId= uploadTextEmbeddingModel();
+            loadModel(modelId);
+            createPipelineProcessor(modelId,PIPELINE_NAME);
+            createIndexWithConfiguration(
+                    testIndex,
+                    Files.readString(Path.of(classLoader.getResource("processor/IndexMappings.json").toURI())),
+                    PIPELINE_NAME
+            );
+            addDocument(testIndex, DOC_ID,TEST_FIELD,TEXT);
+        }else {
+            Map<String,Object> pipeline= getIngestionPipeline(PIPELINE_NAME);
+            assertNotNull(pipeline);
+            String modelId=getModelId(pipeline, TEXT_EMBEDDING_PROCESSOR);
```

Comment:
Why we're retrieving model id from index and not reusing it from the previous step (runningAgainstOldCluster)? Did you check if model id is same or it changes between those two stages of test?

Reply:
When the cluster is running against older version then we create a ingestion pipeline with a text embedding processor in it. That time I add model id into it.


Now when the cluster gets updated with a new version of OS then I cannot persist the model Id in a global variable. Similarly, I cannot store a task Id In which I can persist it and later use it to fetch model Id. So the only way I could fetch model id is by text embedding processor.

Reply:
In Knn they can store model id but for neural we can't.

Reply:
I such case it's possible to have unexpected model id. In neural search integ tests there are some flaky tests where problem is exactly that - model has not been either deployed or deleted properly. We can leave it like this for now, just need to keep this issue in mind when more tests are added in future.

Reply:
I have taken care of it now